src/h5web/providers/hdf5-models.ts

```diff
@@ -128,3 +129,7 @@ interface HDF5CompoundTypeField {
   name: string;
   type: HDF5Type;
 }
+
+interface HDF5UnknownType {
+  class: HDF5TypeClass.Unknown;
+}
```

Comment:
What about adding the rawType in this interface ?

Reply:
You mean instead of having it on the `Dataset` or `Datatype`? The benefit of the current solution is that we can inspect the raw type if we recognise the HSDS type but don't process it correctly. This could be useful for detecting problems with the special types like "boolean as enum" or "complex as compound", or maybe others in the future.

Reply:
I see. Still, it could make more sense to me to add to `HDF5Type` as `raw`, no ?

Reply:
Earlier this afternoon, I would have said yes, because I tried switching to `interface HDF5Type { class: HDF5TypeClass }` along with `extends HDF5Type`. I would then have been able to add a `raw` property to the base class.

However, I ran into a lot of issues with this typing and I had to revert back to `type HDF5Type = HDF5NumericType | HDF5StringType | ...`. And in this case, I'd have to add `raw` to every sub-type...

Reply:
I think having `rawType` on `Dataset` is actually consistent with `rawLink` on `Link`.